x-pack/legacy/plugins/siem/public/lib/connectors/servicenow.tsx

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useCallback, ChangeEvent } from 'react';
+import {
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiFieldPassword,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { isEmpty, get } from 'lodash/fp';
+
+import {
+  ActionConnectorFieldsProps,
+  ActionTypeModel,
+  ValidationResult,
+  ActionParamsProps,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../../../../plugins/triggers_actions_ui/public/types';
+
+import { FieldMapping } from '../../pages/case/components/configure_cases/field_mapping';
+
+import * as i18n from './translations';
+
+import { ServiceNowActionConnector } from './types';
+import { isUrlInvalid } from './validators';
+
+import { connectors, defaultMapping } from './config';
+import { CasesConfigurationMapping } from '../../containers/case/configure/types';
+
+const serviceNowDefinition = connectors['.servicenow'];
+
+interface ServiceNowActionParams {
+  message: string;
+}
+
+export function getActionType(): ActionTypeModel {
+  return {
+    id: serviceNowDefinition.actionTypeId,
+    iconClass: serviceNowDefinition.logo,
+    selectMessage: i18n.SERVICENOW_DESC,
+    actionTypeTitle: i18n.SERVICENOW_TITLE,
+    validateConnector: (action: ServiceNowActionConnector): ValidationResult => {
+      const errors = {
+        apiUrl: [] as string[],
+        username: [] as string[],
+        password: [] as string[],
+      };
+
+      if (!action.config.apiUrl) {
+        errors.apiUrl = [...errors.apiUrl, i18n.SERVICENOW_API_URL_REQUIRED];
+      }
+
+      if (isUrlInvalid(action.config.apiUrl)) {
+        errors.apiUrl = [...errors.apiUrl, i18n.SERVICENOW_API_URL_INVALID];
+      }
+
+      if (!action.secrets.username) {
+        errors.username = [...errors.username, i18n.SERVICENOW_USERNAME_REQUIRED];
+      }
+
+      if (!action.secrets.password) {
+        errors.password = [...errors.password, i18n.SERVICENOW_PASSWORD_REQUIRED];
+      }
+
+      return { errors };
+    },
+    validateParams: (actionParams: ServiceNowActionParams): ValidationResult => {
+      const errors = {
+        message: [] as string[],
+      };
+      return { errors };
```

Comment:
no reason to even make an errors object here right?
```
validateParams: (actionParams: ServiceNowActionParams): ValidationResult => ({ errors: {
        message: []
      }})
```
just not a fan of all these `as` typings. we get the type here for free from `ValidationResult`... right?